NPE in `UnaryOperator::typeChanged`

```
Caused by: java.lang.NullPointerException: Cannot invoke "de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression.setType(de.fraunhofer.aisec.cpg.graph.types.Type, de.fraunhofer.aisec.cpg.graph.HasType)" because "this.input" is null
	at de.fraunhofer.aisec.cpg.graph.statements.expressions.UnaryOperator.typeChanged(UnaryOperator.java:172)
	at de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression.lambda$refreshType$10(Expression.java:211)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression.refreshType(Expression.java:209)
	at de.fraunhofer.aisec.cpg.passes.TypeHierarchyResolver.refreshType(TypeHierarchyResolver.java:114)
	at de.fraunhofer.aisec.cpg.passes.TypeHierarchyResolver.refreshType(TypeHierarchyResolver.java:111)
	at de.fraunhofer.aisec.cpg.passes.TypeHierarchyResolver.refreshType(TypeHierarchyResolver.java:111)
```

--- a/cpg-library/src/main/java/de/fraunhofer/aisec/cpg/graph/statements/expressions/UnaryOperator.java
+++ b/cpg-library/src/main/java/de/fraunhofer/aisec/cpg/graph/statements/expressions/UnaryOperator.java
@@ -116,7 +116,7 @@ public class UnaryOperator extends Expression implements TypeListener {
 
   private boolean getsDataFromInput(TypeListener listener) {
     checked.clear();
-    return input.getTypeListeners().stream().anyMatch(l -> getsDataFromInput(l, listener));
+    return input != null && input.getTypeListeners().stream().anyMatch(l -> getsDataFromInput(l, listener));
   }
 
   public String getOperatorCode() {
@@ -170,7 +170,7 @@ public class UnaryOperator extends Expression implements TypeListener {
         newType = src.getPropagationType().dereference();
       }
 
-      // We are a fuzzy parser, so while this should not happen, there is no garuantee that input is not null
+      // We are a fuzzy parser, so while this should not happen, there is no guarantee that input is not null
       if(input != null) {
         input.setType(newType, this);
       }

--- a/cpg-library/src/main/java/de/fraunhofer/aisec/cpg/graph/statements/expressions/UnaryOperator.java
+++ b/cpg-library/src/main/java/de/fraunhofer/aisec/cpg/graph/statements/expressions/UnaryOperator.java
@@ -169,7 +169,11 @@ public class UnaryOperator extends Expression implements TypeListener {
       } else if (operatorCode.equals("&")) {
         newType = src.getPropagationType().dereference();
       }
-      input.setType(newType, this);
+
+      // We are a fuzzy parser, so while this should not happen, there is no garuantee that input is not null
+      if(input != null) {
+        input.setType(newType, this);
+      }
     }
 
     if (!previous.equals(this.type)) {

